# HG changeset patch
# User Bernhard Walle <bernhard@bwalle.de>
# Date 1399806036 -7200
#      Sun May 11 13:00:36 2014 +0200
# Node ID 5632a171e7c1dc4c0fb5c7a8352afb4188cd29d5
# Parent  e35fa03cd2043496839aa5fa3a1a3d2d375b5535
complibs/ppl: Fix build of ppl 0.11.2 with gcc 4.9

  From: Roberto Bagnara <bagnara@cs.unipr.it>
  Date: Tue, 29 Apr 2014 19:51:43 +0000 (+0200)
  Subject: Added missing inclusions.  Use std::ptrdiff_t.
  X-Git-Url: http://www.cs.unipr.it/git/gitweb.cgi?p=ppl%2Fppl.git;a=commitdiff_plain;h=61d4e14dfd9f1121e9b4521dead5728b2424dd7c

  Added missing inclusions.  Use std::ptrdiff_t.
  (Thanks to Paulo Cesar Pereira de Andrade.)

Signed-off-by: Bernhard Walle <bernhard@bwalle.de>

diff -r e35fa03cd204 -r 5632a171e7c1 patches/ppl/0.11.2/500-ptrdiff_t.patch
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/patches/ppl/0.11.2/500-ptrdiff_t.patch	Sun May 11 13:00:36 2014 +0200
@@ -0,0 +1,95 @@
+From: Bernhard Walle <bernhard@bwalle.de>
+Subject: Fix compilation with gcc 4.9 (ptrdiff_t errors)
+
+This is a backport of following patch in ppl git
+
+  commit 61d4e14dfd9f1121e9b4521dead5728b2424dd7c
+  Author: Roberto Bagnara <bagnara@cs.unipr.it>
+  Date:   Tue Apr 29 21:51:43 2014 +0200
+
+      Added missing inclusions.  Use std::ptrdiff_t.
+      (Thanks to Paulo Cesar Pereira de Andrade.)
+
+---
+ src/Congruence_System.defs.hh     |    3 ++-
+ src/Constraint_System.defs.hh     |    3 ++-
+ src/Generator_System.defs.hh      |    3 ++-
+ src/Grid_Generator_System.defs.hh |    3 ++-
+ 4 files changed, 8 insertions(+), 4 deletions(-)
+
+--- a/src/Grid_Generator_System.defs.hh
++++ b/src/Grid_Generator_System.defs.hh
+@@ -30,6 +30,7 @@ site: http://www.cs.unipr.it/ppl/ . */
+ #include "Variables_Set.types.hh"
+ #include "Grid.types.hh"
+ #include <iosfwd>
++#include <cstddef>
+ 
+ namespace Parma_Polyhedra_Library {
+ 
+@@ -267,7 +268,7 @@ public:
+   class const_iterator
+     : public std::iterator<std::forward_iterator_tag,
+ 			   Grid_Generator,
+-			   ptrdiff_t,
++			   std::ptrdiff_t,
+ 			   const Grid_Generator*,
+ 			   const Grid_Generator&>,
+       private Generator_System::const_iterator {
+--- a/src/Congruence_System.defs.hh
++++ b/src/Congruence_System.defs.hh
+@@ -33,6 +33,7 @@ site: http://www.cs.unipr.it/ppl/ . */
+ #include "Grid.types.hh"
+ #include "Grid_Certificate.types.hh"
+ #include <iosfwd>
++#include <cstddef>
+ 
+ namespace Parma_Polyhedra_Library {
+ 
+@@ -235,7 +236,7 @@ public:
+   class const_iterator
+     : public std::iterator<std::forward_iterator_tag,
+ 			   Congruence,
+-			   ptrdiff_t,
++			   std::ptrdiff_t,
+ 			   const Congruence*,
+ 			   const Congruence&> {
+   public:
+--- a/src/Constraint_System.defs.hh
++++ b/src/Constraint_System.defs.hh
+@@ -33,6 +33,7 @@ site: http://www.cs.unipr.it/ppl/ . */
+ #include "Congruence_System.types.hh"
+ #include <iterator>
+ #include <iosfwd>
++#include <cstddef>
+ 
+ namespace Parma_Polyhedra_Library {
+ 
+@@ -204,7 +205,7 @@ public:
+   class const_iterator
+     : public std::iterator<std::forward_iterator_tag,
+ 			   Constraint,
+-			   ptrdiff_t,
++			   std::ptrdiff_t,
+ 			   const Constraint*,
+ 			   const Constraint&> {
+   public:
+--- a/src/Generator_System.defs.hh
++++ b/src/Generator_System.defs.hh
+@@ -33,6 +33,7 @@ site: http://www.cs.unipr.it/ppl/ . */
+ #include "Polyhedron.types.hh"
+ #include "Poly_Con_Relation.defs.hh"
+ #include <iosfwd>
++#include <cstddef>
+ 
+ namespace Parma_Polyhedra_Library {
+ 
+@@ -250,7 +251,7 @@ public:
+   class const_iterator
+     : public std::iterator<std::forward_iterator_tag,
+ 			   Generator,
+-			   ptrdiff_t,
++			   std::ptrdiff_t,
+ 			   const Generator*,
+ 			   const Generator&> {
+   public:
